source/PowerUp.js

```diff
@@ -97,8 +97,286 @@ export class FiveShoota extends PowerUp {
 export class RapidFire extends PowerUp {
       constructor() {
           super()
-          this.fireRate = 80
-          this.bulletSpeed = 150
+          this.vector = new Victor(1,0)
+          this.bulletSpeed = 2
+          this.projectileType = "bullet"
+          this.rapidFire = true
+      }
+
+      fire(curShip) {
+
+          game.addChild(new Projectile(
+              curShip.position.x + curShip.width,
+              curShip.position.y + curShip.height/2,
+              this.vector, curShip, this.bulletSpeed, this.projectileType))
+      }
+
+}
+
+export class RapidSprayShot extends PowerUp {
+
+      constructor() {
+          super()
+          this.vector = new Victor(1,0)
+          this.vectorN = new Victor(1,0.25)
+          this.vectorP = new Victor(1,-0.25)
+          this.bulletSpeed = 1
+          this.projectileType = "bullet"
+          this.rapidFire = true
+      }
+
+        fire(curShip) {
+
+            this.vector.normalize()
+            this.vectorN.normalize()
+            this.vectorP.normalize()
```

Comment:
Can we not normalize these once in the constructor and then not call it again anymore? Seems a waste to keep normalizing the vectors every single time we fire.